tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -3494,6 +3488,38 @@ def clone_for_deployment(self,
         predictor_clone.save_space()
         return predictor_clone if return_clone else predictor_clone.path
 
+    @staticmethod
+    def _check_if_hyperparameters_handle_text(hyperparameters: dict) -> bool:
+        """Check if hyperparameters contain a model that supports raw text features as input"""
+        models_in_hyperparameters = set()
+        is_advanced_hyperparameter_type = False
+        for key in hyperparameters:
+            if isinstance(key, int) or key == 'default':
+                is_advanced_hyperparameter_type = True
+        if is_advanced_hyperparameter_type:
+            for key in hyperparameters:
+                for m in hyperparameters[key]:
+                    models_in_hyperparameters.add(m)
+        else:
+            for key in hyperparameters:
+                models_in_hyperparameters.add(key)
+        models_in_hyperparameters_raw_text_compatible = []
```

Comment:
```python
for key, m in hyperparameters.items():
    models_in_hyperparameters.add(m if is_advanced_hyperparameter_type else key)
```

Reply:
```
x = {1: {'GBM': {}}}
a = set()
is_advanced_hyperparameter_type = True
for key, m in x.items():
    a.add(m if is_advanced_hyperparameter_type else key)
Traceback (most recent call last):
  File "/Users/neerick/workspace/virtual/autogluon38/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3433, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-8-2d35d45650af>", line 2, in <module>
    a.add(m if is_advanced_hyperparameter_type else key)
TypeError: unhashable type: 'dict
```

The proposed code is not the same and will error.